### PR TITLE
2038: Move map code to map.rb file and add some tiles

### PR DESCRIPTION
**Pull request by jason-johnson**
I have moved the map code to map.rb as other titles seem to have done.  I've also begun adding an initial cut of how 2038 tiles might be modelled.  For now I just take the simplest possible thing that will let me test running actual routes with no concern on how it looks.

I also did make a couple of changes to more closely model how "off board" locations work in 2038.

### CODE REVIEW

File: lib/engine/game/g_2038/map.rb [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+            'count' => 2,
+            'color' => 'yellow',
+            'code' => 'town=revenue:yellow_30|gray_60;town=revenue:20;town=revenue:20;path=a:1,b:_0;path=a:2,b:_0;'\
+                      'path=a:4,b:_1;path=a:5,b:_1;path=a:0,b:_2;path=a:3,b:_2',
+          },
+          '2011' =>
+          {
+            'count' => 1,
+            'color' => 'yellow',
+            'code' => 'city=revenue:yellow_30|gray_60,slots:2;path=a:0,b:_0;path=a:1,b:_0;path=a:2,b:_0;path=a:4,b:_0;'\
```
Comment (michaeljb) [inline lib/engine/game/g_2038/map.rb:84]: Should this and the following tiles also have a `path=a:3,b:_0`?

File: lib/engine/game/g_2038/map.rb [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+                      'path=a:5,b:_0;label=I',
+          },
+          '2012' =>
+          {
+            'count' => 1,
+            'color' => 'yellow',
+            'code' => 'city=revenue:80,slots:4;path=a:0,b:_0;path=a:1,b:_0;path=a:2,b:_0;path=a:4,b:_0;'\
+                      'path=a:5,b:_0;label=I',
```
Comment (michaeljb) [inline lib/engine/game/g_2038/map.rb:92]: Is there any difference between tiles 2012 to 2022?

File: lib/engine/game/g_2038/map.rb (hunk 1 above)
Comment (jason-johnson) [reply lib/engine/game/g_2038/map.rb:84]: Ah yes it should.

File: lib/engine/game/g_2038/map.rb (hunk 2 above)
Comment (jason-johnson) [reply lib/engine/game/g_2038/map.rb:92]: Addressed in comment above.

**Review by michaeljb — APPROVED** [review]

**Pull request merged**